src/index.js

```diff
@@ -18,13 +18,18 @@ const camelize = text => {
   return text.substr(0, 1).toLowerCase() + text.substr(1)
 }
 
-const mapToObjectNotation = props => {
+/**
+ * @template T
+ * @param {T} props
+ * @returns {import("type-fest").CamelCasedPropertiesDeep<T>}
```

Comment:
```suggestion
 * @returns {import("type-fest").CamelCasedPropertiesDeep<T> & import("type-fest").KebabCase<T>}
```
to get both `op.size1` and `op["--size-1"]` to work, to address part of @argyleink's feedback https://github.com/argyleink/open-props/pull/225#pullrequestreview-944673584 

I'll keep investigating [my alternative solution](https://github.com/argyleink/open-props/pull/223), but I'm also liking the approach in this PR 😃

Reply:
Thanks for the suggestion! `T` already infers the kebab keys from the original type, so I just needed to intersect `T` with the camel cased type: https://github.com/argyleink/open-props/commit/bd34e2f39c9add0cf1afc04797cd20c5901b281f